src/components/organisms/Common/AssetModal/hooks.ts

```diff
@@ -0,0 +1,172 @@
+import { useCallback, useState, useEffect } from "react";
+import { useIntl } from "react-intl";
+
+import {
+  AssetsQuery,
+  useAssetsQuery,
+  useCreateAssetMutation,
+  useRemoveAssetMutation,
+  Maybe,
+} from "@reearth/gql";
+import { useNotification } from "@reearth/state";
+
+export type AssetNodes = NonNullable<AssetsQuery["assets"]["nodes"][number]>[];
+
+export enum AssetSortType {
+  Date = "DATE",
+  Name = "NAME",
+  Size = "SIZE",
+}
+
+export default (teamId?: string) => {
+  const intl = useIntl();
+  const [, setNotification] = useNotification();
+  const [sort, setSort] = useState<{ type?: Maybe<AssetSortType>; reverse?: boolean }>();
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const assetsPerPage = 20;
+
+  const pagination = useCallback(
+    (endCursor?: string | null) => {
+      const reverseOrder =
+        (!sort?.type && !sort?.reverse) ||
+        (sort?.type === "DATE" && !sort?.reverse) ||
+        (sort?.type !== "DATE" && sort?.reverse);
+
+      return {
+        after: !reverseOrder ? endCursor : undefined,
+        before: reverseOrder ? endCursor : undefined,
+        first: !reverseOrder ? assetsPerPage : undefined,
+        last: reverseOrder ? assetsPerPage : undefined,
+      };
+    },
+    [sort],
+  );
```

Comment:
Rather than using useCallback, it would be more straightforward to make it a simple function and place it outside the hook.

```ts
function pagination(sort: { type?: Maybe<AssetSortType>; reverse?: boolean } | undefined, endCursor?: string | null) {
  const reverseOrder =
    (!sort?.type && !sort?.reverse) ||
    (sort?.type === "DATE" && !sort?.reverse) ||
    (sort?.type !== "DATE" && sort?.reverse);

  return {
    after: !reverseOrder ? endCursor : undefined,
    before: reverseOrder ? endCursor : undefined,
    first: !reverseOrder ? assetsPerPage : undefined,
    last: reverseOrder ? assetsPerPage : undefined,
  };
}
```

Reply:
This uses useState values inside the hooks. If it was a more generic function that could be used elsewhere I could understand why pulling it out like that would be useful.
Can you explain how it makes it simpler (since the kind of confusing conditional logic needs to remain)?

Reply:
```
function pagination(sortType: AssetSortType | undefined, reverse: boolean | undefined, endCursor?: string | null) {
  const reverseOrder = (!sortType && reverse) || (sortType === "DATE" ? !reverse : reverse);

  return {
    after: !reverseOrder ? endCursor : undefined,
    before: reverseOrder ? endCursor : undefined,
    first: !reverseOrder ? assetsPerPage : undefined,
    last: reverseOrder ? assetsPerPage : undefined,
  };
}
```